src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -370,7 +370,53 @@ def deleteBlocks(self):
           If no => do nothing (check again next cycle)
         """
         # FIXME: figure out the proper logic for rule block deletion
+        # TODO: Update metadata after deleting block rules
         logging.info("Starting deleteBlocks methods --> IMPLEMENT-ME!!!")
+        logging.info("Starting deleteBlocks methods --> I'M ON IT...")
+
+        #Get list of blocks that can be deleted
+        blockDict = self.findDeletableBlocks.execute(transaction=False)
+
+        if not blockDict:
+            logging.info("No blocks were found")
+            return
+
+        deletableBlocks = []
+		# Obtain list of target RSE for each block
+        for blockName in blockDict:
+            rules = self.rucio.getParentContainerRules(name=blockName, scope=self.scope, account=self.rucioAcct)
```

Comment:
For this block of code, I think we could first map the block to a container name, perhaps something like:
```
{"container_name_A": ["block_A", "block_B"], "container_name_B": [], ...}
```
such that we resolve the parent rules only once per cycle, having it only 1 or 1M blocks under the same container.

Even a better improvement would be to keep a containerRules cache (<= 1 hour), which contains parent container names and a list of replication rules. This will only be useful if we run this deleteBlocks method a few times every hour though.

Last but not least, if you provide an `account` keyword arg, then it *should* return only rules matching that account. Which means, your `if` statement on L393 isn't really necessary.

Reply:
I'll start adding the container-blocks mapping. I think deleteBlocks will probably run several times and hour, so caching container rules also seems like a good idea. 

I'll delete the if statement. It was necessary before you added `getParentContainerRules` and I forgot to remove it afterwards.